tests/test_extensions/test_ext_autosummary.py

```diff
@@ -356,6 +356,49 @@ def test_autosummary_generate_content_for_module_imported_members_inherited_modu
     assert context['objtype'] == 'module'
 
 
+@pytest.mark.sphinx(testroot='ext-autosummary')
+def test_autosummary_generate_content_for_module_imported_members_inherited_class(app):
+    import autosummary_dummy_inherited_module
+    template = Mock()
+
+    generate_autosummary_content('autosummary_dummy_inherited_module.InheritedAttrClass',
```

Comment:
For this one, I would like to have a much wider test coverage. More precisely, I want to be able to test the behaviour of private names, mangled names, inherited methods that, static methods. For that, I would usggest creating a separate dummy module with multiple classes inside. I also want to see how multiple inheritance is handled and how shadowing names are done.

Most of the time, our tests cover simple cases which are fine but everytime there is an issue with autodoc, it's always because of some weird constructions. So I'd like to at least cover non-trivial cases (without trying really pathological cases).

Reply:
Sure thing, I might need to lean a bit about these more niche cases but happy to give it a first go!

Reply:
Have made a module where I have tried to test some of these features. Let me know if there is anything specific/additional that you think is missing.